src/pages/OfferNew/Introduction/Sidebar/DetailsModal/index.tsx

```diff
@@ -0,0 +1,425 @@
+                            formikProps={formikProps}
+                            nameRoot="house"
+                            name="ancillarySpace"
+                          />
+                        </InputGroupRow>
+
+                        <InputGroupRow>
+                          <DetailInput
+                            field={fieldSchema.house.numberOfBathrooms}
+                            formikProps={formikProps}
+                            nameRoot="house"
+                            name="numberOfBathrooms"
+                          />
+                          <DetailInput
+                            field={fieldSchema.house.yearOfConstruction}
+                            formikProps={formikProps}
+                            nameRoot="house"
+                            name="yearOfConstruction"
+                          />
+                        </InputGroupRow>
+
+                        <DetailInput
+                          field={fieldSchema.house.householdSize}
+                          formikProps={formikProps}
+                          nameRoot="house"
+                          name="householdSize"
+                        />
+                        <DetailInput
+                          field={fieldSchema.house.isSubleted}
+                          formikProps={formikProps}
+                          nameRoot="house"
+                          name="isSubleted"
+                        />
+                      </InputGroup>
+                      <SupportSection onButtonClick={() => Intercom('show')} />
+                    </ContentColumn>
+                    <ContentColumn>
+                      <FieldArray
+                        name="house.extraBuildings"
+                        render={(arrayHelpers) => (
+                          <>
+                            <ContentColumnTitle>
+                              {textKeys.DETAILS_MODULE_EXTRABUILDINGS_TABLE_TITLE()}
+                              <ContentColumnTitleButton
+                                type="button"
+                                onClick={() => {
+                                  const defaultExtraBuilding: ExtraBuildingInput = {
+                                    type: ExtraBuildingType.Garage,
+                                    area: 10,
+                                    hasWaterConnected: false,
+                                  }
+                                  arrayHelpers.insert(0, defaultExtraBuilding)
+                                }}
+                              >
+                                {textKeys.DETAILS_MODULE_EXTRABUILDINGS_TABLE_BUTTON()}
+                              </ContentColumnTitleButton>
+                            </ContentColumnTitle>
+
+                            {formikProps.values.house?.extraBuildings?.map(
+                              (_, index) => (
+                                <InputGroup key={index}>
```

Comment:
i believe this (`key=index`) might cause issues if you remove a house that isn't the last, considering React sees the `key` as the "definition" of which DOM element this is and the id in the middle remains the same. I guess since we're updating the name though (ie `extraBuildings.${index}.type`) this might not be an issue but it'll most likely cause strange and hard-to-track-down bugs in the future if we come back here. Maybe we could find a better `key` prop? just like a hash, id, anything except the array index

Reply:
honestly, this sucks but fixing it will suck even more 😂 i'll leave it for now